tests/test_schema.py

```diff
@@ -1200,11 +1200,33 @@ def test_promote_float_to_double() -> None:
     assert isinstance(applied.fields[0].field_type, DoubleType)
 
 
+def test_promote_long_to_int() -> None:
```

Comment:
```suggestion
def test_allow_union_long_with_int() -> None:
```
this isn't a type promotion, right? we're testing `union_by_name` now allows type-widening

Reply:
Good call, updated 👍